docs/DeveloperGuide.md

```diff
@@ -2,12 +2,343 @@
+spending. This information was used by other classes to perform their component tasks.
+
+### Storage Component
+#### Motivation behind the component:
++ Allows the user to save changes, so that they can resume where they left off.
++ Allows advanced users to edit files directly, enabling fast, manual adjustments.
+
+#### How the Storage Component works:
++ Variables and File Path:
+    + `incomes` → `./incomes.txt`
+    + `spendings` → `./spendings.txt`
+    + `password` → `./password.txt`
+
++ To save edited lists:
+    + It is done when users type `bye` or after keyboard interrupts (i.e.Ctrl-c), which signals the end of the program.
+    + The lists are saved to a user-editable format in their respective files.
+
++ To load saved lists:
+  + It is done upon program startup, when `Wiagi` is constructed.
+  + Within the `Wiagi` constructor, it will create a new instance of `Storage`, which will then load the data at the 
+  `incomes` and `spendings` file paths to an `IncomeList` and `SpendingList` respectively.
+  + `Wiagi` will then retrieve the lists in `Storage` to initialise its lists.
+  + Data corruption in the file triggers an exception, often due to user-editing errors.
+  + For missing files (e.g., new users), files are created and the initialised lists will be empty.
+
++ To load password:
+  + The hashed password will simply be loaded from the password file.
+  + For missing files (e.g., new users), users will be prompted to set a new password at the start of the
+  program. The entered password will then be hashed and stored in a newly created password file.
+
+#### Implementation:
+#### Storage class
+The `Storage` class is a class that stores `incomes`, `spendings` and `password`. 
+Upon instantiation, it will call `IncomeListStorage.load()`, `SpendingListStorage.load()` and `LoginStorage.load()`, 
+which will initialise the variables in `Storage` respectively.
+
+#### save method in `IncomeListStorage` `SpendingListStorage`
+<img src="./Diagrams/Storage/saveStorageSequenceDiagram.png" alt="saveStorageSequenceDiagram" width="600" height="400"/><br>
+Both classes have similar implementation for `save()`, except that `SpendingListStorage` saves budget details in the 
+first line of its respective text file.
++ Format: `daily budget | monthly budget | yearly budget`
++ A for loop will loop through the list, and get each of the attributes of each entry within it and separate them by 
+`|`. Hence, each entry will be written line by line to the file.
++ Format: `amount | description | date | tag | recurrence frequency | last recurrence date | last recurrence day`
+  + E.g. `add income 10 part time /2024-10-10/ *job* ~monthly~` will be stored as
+    `10.0|part time|2024-10-10|job|MONTHLY|2024-10-10|10`
+
+#### load method in `IncomeListStorage` `SpendingListStorage`
+<img src="./Diagrams/Storage/loadStorageSequenceDiagram.png" alt="loadStorageSequenceDiagram" width="600" height="400"/><br>
```

Comment:
Would there be a self-invocation for load() methods within the Storages?

Perhaps use UML notation for sequence diagrams instead of having "labels" to state what it is doing.
This problem exists in multiple sequence diagrams.